js/misc/paygManager.js

```diff
@@ -88,6 +110,27 @@ function _isPaygEnabled() {
     return false;
 }
 
+// Takes an UNIX timestamp (in seconds) and returns a string
+// with a precision level appropriate to show to the user.
+function timeToString(seconds) {
+    if (seconds < 60)
+        return Gettext.ngettext("%s second", "%s seconds", seconds).format(seconds);
+
+    let minutes = Math.ceil(seconds / 60);
+    if (minutes < 60)
+        return Gettext.ngettext("%s minute", "%s minutes", minutes).format(minutes);
+
+    let hours = Math.floor(minutes / 60);
+    let minutesPast = minutes % 60;
+
+    if (minutesPast == 0)
+        return Gettext.ngettext("%s hour", "%s hours", hours).format(hours);
+
+    // For anything that is not a round number of hours, we resort to the
+    // number of minutes, which is simpler than mixing hours and minutes.
+    return "%s minutes".format(minutes);
```

Comment:
Given my suggestion to add a one-day warning, the user could boot with something like 15 hours and 39 minutes.  In that case, we should round down to 15 hours rather than display 939 minutes.  I'd be okay with showing full minutes for times less than 2 hours, but for times above 2 hours I would just round down to the nearest number of hours.

Reply:
Sounds good, I'll change that then.